### PR TITLE
Add unit tests into tst_value.cpp and tst_prg.cpp

### PR DESCRIPTION
**Pull request by EthanPadden**

### CODE REVIEW

File: src/test/tst_value.cpp [hunk 1]
```diff
@@ -89,44 +70,148 @@ void TestValue::tst_compare()
...
+    v = Value(false);
+
+    //Testing validity
+    QVERIFY(v.isValid() == true);
+
+    //Testing type
+    QVERIFY(v.type() == Value::BOOL);
+
+    //Testing value
+    QVERIFY(v.toBool() == false);
```
Comment (cardinot) [inline src/test/tst_value.cpp]: just for consistency, as we're retrieving the value, it's better to use QCOMPARE
``` c
QCOMPARE(v.toBool(), false);
```

File: src/test/tst_value.cpp [hunk 2]
```diff
@@ -89,44 +70,148 @@ void TestValue::tst_compare()
...
+
+    //Testing validity
+    QVERIFY(v.isValid() == true);
 
-    // compare bool
+    //Testing type
+    QVERIFY(v.type() == Value::BOOL);
+
+    //Testing value
+    QVERIFY(v.toBool() == true);
```
Comment (cardinot) [inline src/test/tst_value.cpp]: just for consistency, as we're retrieving the value, it's better to use QCOMPARE
``` c
QCOMPARE(v.toBool(), true);
```

File: src/test/tst_value.cpp [hunk 3]
```diff
@@ -89,44 +70,148 @@ void TestValue::tst_compare()
...
+
+    //Testing validity
+    QVERIFY(v.isValid() == true);
+
+    //Testing type
+    QVERIFY(v.type() == Value::BOOL);
+
+    //Testing value
+    QVERIFY(v.toBool() == false);
+
```
Comment (cardinot) [inline src/test/tst_value.cpp]:
``` c
QCOMPARE(v.toQString(), QString("0"));
```

File: src/test/tst_value.cpp [hunk 4]
```diff
@@ -89,44 +70,148 @@ void TestValue::tst_compare()
...
+    Value v;
+
+    //VALUE TESTS
+    v = Value(4.5123788);
+
+    //Testing validity
+    QVERIFY(v.isValid() == true);
+
+    //Testing type
+    QVERIFY(v.type() == Value::DOUBLE);
```
Comment (cardinot) [inline src/test/tst_value.cpp]:
``` c
QVERIFY(v.isDouble());
QVERIFY(!v.isBool());
QVERIFY(!v.isChar());
QVERIFY(!v.isInt());
QVERIFY(!v.isString());
```

File: src/test/tst_value.cpp [hunk 5]
```diff
@@ -89,44 +70,148 @@ void TestValue::tst_compare()
...
+    Value v;
+
+    //VALUE TESTS
+    v = Value(-10);
+
+    //Testing validity
+    QVERIFY(v.isValid() == true);
+
+    //Testing type
+    QVERIFY(v.type() == Value::INT);
```
Comment (cardinot) [inline src/test/tst_value.cpp]:
``` c
QVERIFY(v.isInt());
QVERIFY(!v.isBool());
QVERIFY(!v.isChar());
QVERIFY(!v.isDouble());
QVERIFY(!v.isString());
```

File: src/test/tst_value.cpp [hunk 6]
```diff
@@ -89,44 +70,148 @@ void TestValue::tst_compare()
...
     QCOMPARE(v1 >  v2, true);
     QCOMPARE(v1 <  v2, false);
+}
+void TestValue::tst_valueChar(){
+    Value v;
 
-    // compare int
-    v1 = Value(100), v2 = Value(50);
+    //VALUE TESTS
+    const char* chr = "abc£ãã&";
```
Comment (cardinot) [inline src/test/tst_value.cpp]: a `const char*` is actually a string
you should be testing a simple `char` here, for example:
``` c
char chr = 'a';
```

File: src/test/tst_value.cpp [hunk 7]
```diff
@@ -89,44 +70,148 @@ void TestValue::tst_compare()
...
+    v = Value(chr);
+
+    //Testing validity
+    QVERIFY(v.isValid() == true);
+
+    //Testing type
+    //Verify that it is a string, as chr is a character pointer.
+    QVERIFY(v.type() == Value::STRING);
```
Comment (cardinot) [inline src/test/tst_value.cpp]: fix the type:
``` c
QVERIFY(v.type() == Value::CHAR);
```
and add the isXXX block:
``` c
QVERIFY(v.isChar());
QVERIFY(!v.isBool());
QVERIFY(!v.isDouble());
QVERIFY(!v.isInt());
QVERIFY(!v.isString());
```

File: src/test/tst_value.cpp [hunk 8]
```diff
@@ -89,44 +70,148 @@ void TestValue::tst_compare()
...
+
+    //Testing value
+    QCOMPARE(v.toString(), chr);
```
Comment (cardinot) [inline src/test/tst_value.cpp]: fix it to ` QCOMPARE(v.toChar(), chr);`

**Comment by EthanPadden** [comment]
Thank you for the review, I'll make those changes.

**Comment by cardinot** [comment]
Thanks @EthanPadden don't forget to remove the `CMakeLists.txt.user` (as in PR #11)

**Pull request merged**